File: PTT-GC-SPA/src/app/core/models/progressModel.ts
import { InvestmentCost } from './investmentCost';
import { ProgressDetail } from './progressDetail';

export interface ProgressModel {
  initiativeId: number,
  appropriationNo: number,
  wbsNo: string,
  standardProjectDef: string,
  responsible: string,
  solomonCategory: string,
  areaPlant: string,
  physicalBu: string,
  physicalUnit: string,
  details: Array<ProgressDetail>,
  progressHeaderId: number,
  planCost: InvestmentCost,
  actualCost: InvestmentCost,
  // bsc
  // engineering: string,
  // construction: string,
  // procurement: string,
  // commissioningStartup: string,
  // projectManagement: string,
  // riskAndConcern: string,
  // mitigationPlan: string,
  // executiveSummary: string,
  // workForNextMonth: string,
  // environmentKpi: string
}

export interface OutstandingModel {
  id: number;
  year: number;
  month: string;
  initiativeId: number;
  budgetBaht: number;
  actualBaht: number;
  prItemBaht: number;
  poItemBaht: number;
  commitmentBaht: number;
  contingency: number;
  estimateAtCompletion: number;
  isDeleted: boolean;
  outstandingFormArray: OutstandingData[];
}

export interface OutstandingData {
  id: number;
  initiativeId: number;
  year: number;
  month: string;
  outstandingId: number;
  itemDescription: string;
  itemListValueBaht: number;
  rpcDescription: string;
  rpcValueBaht: number;
  outstanding: number;
  isDeleted: boolean;
}

File: PTT-GC-API/Models/ProgressAndMilestone/ProgressModel.cs
using PTT_GC_API.Dtos.ProgressAndMilestone;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace PTT_GC_API.Models.ProgressAndMilestone
{
    public class ProgressModel
    {
        [Key]
        public int ProgressHeaderId { get; set; }
        public int InitiativeId { get; set; }
        public string AppropriationNo { get; set; }
        public string WbsNo { get; set; }
        public string StandardProjectDef { get; set; }
        public string Responsible { get; set; }
        public string SolomonCategory { get; set; }
        public string AreaPlant { get; set; }
        public string PhysicalBu { get; set; }
        public string PhysicalUnit { get; set; }
        public List<ProgressDetail> Details { get; set; }
        public InvestmentCost PlanCost { get; set; }
        public InvestmentCost ActualCost { get; set; }

        // BSC
        public string Engineering { get; set; }
        public string Construction { get; set; }
        public string Procurement { get; set; }
        public string CommissioningStartup { get; set; }
        public string ProjectManagement { get; set; }
        public string RiskAndConcern { get; set; }
        public string MitigationPlan { get; set; }
        public string ExecutiveSummary { get; set; }
        public string WorkForNextMonth { get; set; }
        public string EnvironmentKpi { get; set; }
    }
}
